crates/contract/tests/sandbox/integration_update_votes_cleanup_after_resharing.rs

```diff
@@ -0,0 +1,191 @@
+#[tokio::test]
+async fn test_update_votes_cleared_after_resharing() -> Result<()> {
+    let (_, contract, env_accounts, _) =
+        init_env(&[SignatureScheme::Secp256k1], PARTICIPANT_LEN).await;
+
+    let initial_participants = assert_running_return_participants(&contract).await?;
+    let threshold = assert_running_return_threshold(&contract).await;
+
+    // Propose update and have first 2 participants vote on it
+    let update_id: UpdateId = env_accounts[0]
+        .call(contract.id(), "propose_update")
+        .args_borsh(ProposeUpdateArgs {
+            code: Some(vec![1u8; 1000]),
+            config: None,
+        })
+        .deposit(near_workspaces::types::NearToken::from_near(1))
+        .max_gas()
+        .transact()
+        .await?
+        .json()?;
+
+    for account in &env_accounts[0..2] {
+        account
+            .call(contract.id(), "vote_update")
+            .args_json(json!({"id": update_id}))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+    }
+
+    let proposals_before: dtos::ProposedUpdates =
+        contract.view("proposed_updates").await?.json()?;
+    assert_eq!(proposals_before.0.len(), 1);
+
+    let voters_before: HashSet<_> = proposals_before.0[0]
+        .votes
+        .iter()
+        .map(|v| v.0.parse::<AccountId>().unwrap())
+        .collect();
+    assert_eq!(
+        voters_before,
+        HashSet::from([
+            env_accounts[0].id().as_v2_account_id(),
+            env_accounts[1].id().as_v2_account_id()
+        ])
+    );
+
+    // Reshare with threshold participants, excluding participant 0 who voted
+    let mut new_participants = Participants::new();
+    for (account_id, participant_id, participant_info) in initial_participants
+        .participants()
+        .iter()
+        .skip(1) // Skip participant 0, so participant 1-6 are included
+        .take(threshold.value() as usize)
+    {
+        new_participants
+            .insert_with_id(
+                account_id.clone(),
+                participant_info.clone(),
+                participant_id.clone(),
+            )
+            .map_err(|e| anyhow::anyhow!("Failed to insert participant: {}", e))?;
+    }
+
+    let new_threshold_parameters = ThresholdParameters::new(new_participants, threshold.clone())
+        .map_err(|e| anyhow::anyhow!("{}", e))?;
+    let prospective_epoch_id = EpochId::new(6);
+
+    // Vote for new parameters (skip participant 0, use participants 1-6)
+    for account in &env_accounts[1..threshold.value() as usize + 1] {
+        account
+            .call(contract.id(), "vote_new_parameters")
+            .args_json(json!({
+                "prospective_epoch_id": prospective_epoch_id,
+                "proposal": new_threshold_parameters,
+            }))
+            .max_gas()
+            .transact()
+            .await?
+            .into_result()?;
+    }
```

Comment:
now I realized that using `execute_async_transactions` here will make the test much faster! maybe we should use that for every contract call that we make in a cycle